cloud/services/compute/firewalls.go

```diff
@@ -62,10 +62,12 @@ func (s *Service) DeleteFirewalls() error {
 	for name := range s.scope.Network().FirewallRules {
 		op, err := s.firewalls.Delete(s.scope.Project(), name).Do()
 		if err != nil {
-			return errors.Wrapf(err, "failed to delete forwarding rules")
+			if !gcperrors.IsNotFound(err) {
+				return errors.Wrapf(err, "failed to delete firewall")
+			}
 		}
 		if err := wait.ForComputeOperation(s.scope.Compute, s.scope.Project(), op); err != nil {
```

Comment:
I'm wondering if the call to `wait.ForComputeOperation()` here should be wrapped with a conditional to avoid calling this if the resource was not found above:

```go
if !gcperrors.IsNotFound(err) {
...
}
```

similar with the other files.